[T-8093][14.0][FIX] ecommerce_connector: customer search

manuelregidor opened the pull request:
This PR includes the option "Contact Search Partner Type" in the ecommerce connection configuration. If checked, when searching for customers the fact that the contact is an individual or a business will be added as a new condition to be met.
The search domain has also been extracted into a new method so it is easy to inherit in case further conditions have to be added.

@HaraldPanten @luis-ron

manuelregidor committed:
[T-8093][FIX] ecommerce_connector: avoid searching by partner type to avoid problems finding contacts

Tisho99 reviewed (approved):
LGTM

manuelregidor commented:
@ValentinVinagre

ValentinVinagre commented on ecommerce_connector/models/ecommerce_connection.py:
```diff
@@ -53,6 +53,10 @@ def _selection_languages(self):
         default="ecommerce_id",
         required=True
     )
+    contact_search_partner_type = fields.Boolean(
```
Couldn't it be true by default?

manuelregidor replied:
I would't set it True by default as it might duplicate contacts.

ValentinVinagre reviewed (changes requested):
Please explain the solution in the commit message and expand the readme.

manuelregidor commented:
@ValentinVinagre I've added the changes in the first message of the PR and updated the handbook (https://www.sygel.es/web#id=29&menu_id=996&cids=1&action=1565&model=document.page&view_type=form).

ValentinVinagre commented:
> @ValentinVinagre I've added the changes in the first message of the PR and updated the handbook (https://www.sygel.es/web#id=29&menu_id=996&cids=1&action=1565&model=document.page&view_type=form).

To change:
This PR go 14.0 branch, the name of PR indicate 17.0. 
Explain the error in the commit .

manuelregidor commented:
@ValentinVinagre Changes done

ValentinVinagre commented:
@luis-ron , can you check that everything is working in the client test environment? @manuelregidor will remain idle until this check is made.

Tisho99 commented:
This was done in the https://github.com/sygel-technology/sy-general-connector/pull/16
